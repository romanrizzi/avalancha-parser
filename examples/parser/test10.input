
-- Implicaci√≥n

check A imp B 
check A imp B imp C
check A imp B imp C imp D
check not A imp not B imp not C imp not D

check A1==A2 imp B1==B2
check A1==A2 imp B1==B2 imp C1==C2
check A1==A2 imp B1==B2 imp C1==C2 imp D1==D2
check not A1==A2 imp not B1==B2 imp not C1==C2 imp not D1==D2

check A imp B 
check (A imp B) imp C
check A imp (B imp C) imp D
check not ((A imp not (B imp not C)) imp not D)

check A1==A2 imp B1==B2
check A1==A2 imp (B1==B2 imp C1==C2)
check ((A1==A2 imp B1==B2) imp C1==C2) imp D1==D2
check not (A1==A2 imp not (B1==B2 imp not (C1==C2 imp not (D1==D2))))

