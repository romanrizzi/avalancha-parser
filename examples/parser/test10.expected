[
   "program",
   [],
   [
      [
         "check",
         [
            "imp",
            [
               "equal",
               [
                  "cons",
                  "A",
                  []
               ],
               [
                  "cons",
                  "True",
                  []
               ]
            ],
            [
               "equal",
               [
                  "cons",
                  "B",
                  []
               ],
               [
                  "cons",
                  "True",
                  []
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "imp",
            [
               "equal",
               [
                  "cons",
                  "A",
                  []
               ],
               [
                  "cons",
                  "True",
                  []
               ]
            ],
            [
               "imp",
               [
                  "equal",
                  [
                     "cons",
                     "B",
                     []
                  ],
                  [
                     "cons",
                     "True",
                     []
                  ]
               ],
               [
                  "equal",
                  [
                     "cons",
                     "C",
                     []
                  ],
                  [
                     "cons",
                     "True",
                     []
                  ]
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "imp",
            [
               "equal",
               [
                  "cons",
                  "A",
                  []
               ],
               [
                  "cons",
                  "True",
                  []
               ]
            ],
            [
               "imp",
               [
                  "equal",
                  [
                     "cons",
                     "B",
                     []
                  ],
                  [
                     "cons",
                     "True",
                     []
                  ]
               ],
               [
                  "imp",
                  [
                     "equal",
                     [
                        "cons",
                        "C",
                        []
                     ],
                     [
                        "cons",
                        "True",
                        []
                     ]
                  ],
                  [
                     "equal",
                     [
                        "cons",
                        "D",
                        []
                     ],
                     [
                        "cons",
                        "True",
                        []
                     ]
                  ]
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "imp",
            [
               "not",
               [
                  "equal",
                  [
                     "cons",
                     "A",
                     []
                  ],
                  [
                     "cons",
                     "True",
                     []
                  ]
               ]
            ],
            [
               "imp",
               [
                  "not",
                  [
                     "equal",
                     [
                        "cons",
                        "B",
                        []
                     ],
                     [
                        "cons",
                        "True",
                        []
                     ]
                  ]
               ],
               [
                  "imp",
                  [
                     "not",
                     [
                        "equal",
                        [
                           "cons",
                           "C",
                           []
                        ],
                        [
                           "cons",
                           "True",
                           []
                        ]
                     ]
                  ],
                  [
                     "not",
                     [
                        "equal",
                        [
                           "cons",
                           "D",
                           []
                        ],
                        [
                           "cons",
                           "True",
                           []
                        ]
                     ]
                  ]
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "imp",
            [
               "equal",
               [
                  "cons",
                  "A1",
                  []
               ],
               [
                  "cons",
                  "A2",
                  []
               ]
            ],
            [
               "equal",
               [
                  "cons",
                  "B1",
                  []
               ],
               [
                  "cons",
                  "B2",
                  []
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "imp",
            [
               "equal",
               [
                  "cons",
                  "A1",
                  []
               ],
               [
                  "cons",
                  "A2",
                  []
               ]
            ],
            [
               "imp",
               [
                  "equal",
                  [
                     "cons",
                     "B1",
                     []
                  ],
                  [
                     "cons",
                     "B2",
                     []
                  ]
               ],
               [
                  "equal",
                  [
                     "cons",
                     "C1",
                     []
                  ],
                  [
                     "cons",
                     "C2",
                     []
                  ]
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "imp",
            [
               "equal",
               [
                  "cons",
                  "A1",
                  []
               ],
               [
                  "cons",
                  "A2",
                  []
               ]
            ],
            [
               "imp",
               [
                  "equal",
                  [
                     "cons",
                     "B1",
                     []
                  ],
                  [
                     "cons",
                     "B2",
                     []
                  ]
               ],
               [
                  "imp",
                  [
                     "equal",
                     [
                        "cons",
                        "C1",
                        []
                     ],
                     [
                        "cons",
                        "C2",
                        []
                     ]
                  ],
                  [
                     "equal",
                     [
                        "cons",
                        "D1",
                        []
                     ],
                     [
                        "cons",
                        "D2",
                        []
                     ]
                  ]
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "imp",
            [
               "not",
               [
                  "equal",
                  [
                     "cons",
                     "A1",
                     []
                  ],
                  [
                     "cons",
                     "A2",
                     []
                  ]
               ]
            ],
            [
               "imp",
               [
                  "not",
                  [
                     "equal",
                     [
                        "cons",
                        "B1",
                        []
                     ],
                     [
                        "cons",
                        "B2",
                        []
                     ]
                  ]
               ],
               [
                  "imp",
                  [
                     "not",
                     [
                        "equal",
                        [
                           "cons",
                           "C1",
                           []
                        ],
                        [
                           "cons",
                           "C2",
                           []
                        ]
                     ]
                  ],
                  [
                     "not",
                     [
                        "equal",
                        [
                           "cons",
                           "D1",
                           []
                        ],
                        [
                           "cons",
                           "D2",
                           []
                        ]
                     ]
                  ]
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "imp",
            [
               "equal",
               [
                  "cons",
                  "A",
                  []
               ],
               [
                  "cons",
                  "True",
                  []
               ]
            ],
            [
               "equal",
               [
                  "cons",
                  "B",
                  []
               ],
               [
                  "cons",
                  "True",
                  []
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "imp",
            [
               "imp",
               [
                  "equal",
                  [
                     "cons",
                     "A",
                     []
                  ],
                  [
                     "cons",
                     "True",
                     []
                  ]
               ],
               [
                  "equal",
                  [
                     "cons",
                     "B",
                     []
                  ],
                  [
                     "cons",
                     "True",
                     []
                  ]
               ]
            ],
            [
               "equal",
               [
                  "cons",
                  "C",
                  []
               ],
               [
                  "cons",
                  "True",
                  []
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "imp",
            [
               "equal",
               [
                  "cons",
                  "A",
                  []
               ],
               [
                  "cons",
                  "True",
                  []
               ]
            ],
            [
               "imp",
               [
                  "imp",
                  [
                     "equal",
                     [
                        "cons",
                        "B",
                        []
                     ],
                     [
                        "cons",
                        "True",
                        []
                     ]
                  ],
                  [
                     "equal",
                     [
                        "cons",
                        "C",
                        []
                     ],
                     [
                        "cons",
                        "True",
                        []
                     ]
                  ]
               ],
               [
                  "equal",
                  [
                     "cons",
                     "D",
                     []
                  ],
                  [
                     "cons",
                     "True",
                     []
                  ]
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "not",
            [
               "imp",
               [
                  "imp",
                  [
                     "equal",
                     [
                        "cons",
                        "A",
                        []
                     ],
                     [
                        "cons",
                        "True",
                        []
                     ]
                  ],
                  [
                     "not",
                     [
                        "imp",
                        [
                           "equal",
                           [
                              "cons",
                              "B",
                              []
                           ],
                           [
                              "cons",
                              "True",
                              []
                           ]
                        ],
                        [
                           "not",
                           [
                              "equal",
                              [
                                 "cons",
                                 "C",
                                 []
                              ],
                              [
                                 "cons",
                                 "True",
                                 []
                              ]
                           ]
                        ]
                     ]
                  ]
               ],
               [
                  "not",
                  [
                     "equal",
                     [
                        "cons",
                        "D",
                        []
                     ],
                     [
                        "cons",
                        "True",
                        []
                     ]
                  ]
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "imp",
            [
               "equal",
               [
                  "cons",
                  "A1",
                  []
               ],
               [
                  "cons",
                  "A2",
                  []
               ]
            ],
            [
               "equal",
               [
                  "cons",
                  "B1",
                  []
               ],
               [
                  "cons",
                  "B2",
                  []
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "imp",
            [
               "equal",
               [
                  "cons",
                  "A1",
                  []
               ],
               [
                  "cons",
                  "A2",
                  []
               ]
            ],
            [
               "imp",
               [
                  "equal",
                  [
                     "cons",
                     "B1",
                     []
                  ],
                  [
                     "cons",
                     "B2",
                     []
                  ]
               ],
               [
                  "equal",
                  [
                     "cons",
                     "C1",
                     []
                  ],
                  [
                     "cons",
                     "C2",
                     []
                  ]
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "imp",
            [
               "imp",
               [
                  "imp",
                  [
                     "equal",
                     [
                        "cons",
                        "A1",
                        []
                     ],
                     [
                        "cons",
                        "A2",
                        []
                     ]
                  ],
                  [
                     "equal",
                     [
                        "cons",
                        "B1",
                        []
                     ],
                     [
                        "cons",
                        "B2",
                        []
                     ]
                  ]
               ],
               [
                  "equal",
                  [
                     "cons",
                     "C1",
                     []
                  ],
                  [
                     "cons",
                     "C2",
                     []
                  ]
               ]
            ],
            [
               "equal",
               [
                  "cons",
                  "D1",
                  []
               ],
               [
                  "cons",
                  "D2",
                  []
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "not",
            [
               "imp",
               [
                  "equal",
                  [
                     "cons",
                     "A1",
                     []
                  ],
                  [
                     "cons",
                     "A2",
                     []
                  ]
               ],
               [
                  "not",
                  [
                     "imp",
                     [
                        "equal",
                        [
                           "cons",
                           "B1",
                           []
                        ],
                        [
                           "cons",
                           "B2",
                           []
                        ]
                     ],
                     [
                        "not",
                        [
                           "imp",
                           [
                              "equal",
                              [
                                 "cons",
                                 "C1",
                                 []
                              ],
                              [
                                 "cons",
                                 "C2",
                                 []
                              ]
                           ],
                           [
                              "not",
                              [
                                 "equal",
                                 [
                                    "cons",
                                    "D1",
                                    []
                                 ],
                                 [
                                    "cons",
                                    "D2",
                                    []
                                 ]
                              ]
                           ]
                        ]
                     ]
                  ]
               ]
            ]
         ]
      ]
   ]
]
